src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Centipede engine interface."""
+
+from pathlib import Path
+import re
+
+from clusterfuzz._internal.bot.fuzzers import dictionary_manager
+from clusterfuzz._internal.bot.fuzzers import engine_common
+from clusterfuzz._internal.bot.fuzzers import utils as fuzzer_utils
+from clusterfuzz._internal.system import environment
+from clusterfuzz._internal.system import new_process
+from clusterfuzz.fuzz import engine
+
+_CLEAN_EXIT_SECS = 10
+_TIMEOUT = 1200
```

Comment:
This is quite an increase over libFuzzer's 25 seconds. Any reason for this massive increase? This is the timeout for an individual testcase right? And not the entire fuzzing run?

Also, does Centipede properly report timeouts in a way that we can distinguish from regualr crashes?

Reply:
Ah, I should have used 25: It is simply copied from FuzzBench.
Yes, this is the timeout of an individual testcase, there is no reason that it should be that long.

Re. timeout reports:
I don't recall any doc, let me test it manually.
If it does, I will make it the same as [how libFuzzer handles `timeout` crashes](https://github.com/google/clusterfuzz/blob/c5d3476ef597addbfc8848b2c0ae1be1e428a55c/src/clusterfuzz/_internal/bot/fuzzers/libfuzzer.py#L107).

Reply:
OK, I can confirm that `Centipede` saves timeout inputs to `crash`, just like other crashes (asan, etc.).
One way to distinguish them is via parsing the `Centipede`'s stdout and looking for this message:
```
Timeout of 25 seconds exceeded; exiting
```

Reply:
nice! we need to mature sure both timeouts and OOMs can be properly parsed by stack_Analyzer with this new format. This can come in another PR.

Reply:
OK, then I think all comments are addressed, I'll push now.